server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -230,13 +230,8 @@ public ClusterState execute(ClusterState currentState) {
                 List<String> indices = Arrays.asList(indexNameExpressionResolver.concreteIndexNames(currentState,
                     request.indicesOptions(), true, request.indices()));
 
-                Map<String, DataStream> allDataStreams = currentState.metadata().dataStreams();
-                List<String> dataStreams;
-                if (request.includeGlobalState()) {
-                    dataStreams = new ArrayList<>(allDataStreams.keySet());
-                } else {
-                    dataStreams = indexNameExpressionResolver.dataStreamNames(currentState, request.indicesOptions(), request.indices());
-                }
+                final List<String> dataStreams =
```

Comment:
Just like for indices we should respect the data streams constraints given by the request and not simply always snapshot all data-streams when doing a global snapshot.